lib/librato-services/output.rb

```diff
@@ -143,6 +143,28 @@ def calculate_duration(measurement)
         end
       end
 
+      def sms_message
+        if valid_sms?
+          violations_message
+        else
+          truncated_violations_message
+        end
+      end
+
+      def valid_sms?
+        violations_message.length <= 140
+      end
+
+      def violations_message
+        violations.flat_map do |source, measurements|
+          measurements.map { |measurement| format_measurement(measurement, source) }
+        end.join('. ')
+      end
+
+      def truncated_violations_message
+        "#{violations_message[0..136]}..."
```

Comment:
@akahn didn't realize there's an existing dependency on [`ActiveSupport`](https://github.com/librato/librato-services/blob/master/Gemfile#L7), makes sense to just use core ext in 151300839717f390dfb536b356436844a4646d32.